js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -154,18 +156,30 @@ version flag it can use to trigger a re-render after changes are saved, so chang
 namespaces changes the returned data immediately.
 */
 export function useExecutionSessionStorage(
-  repositoryName: string,
-  pipelineName: string,
+  repoAddress: RepoAddress,
+  pipelineOrJobName: string,
   initial: Partial<IExecutionSession> = {},
 ): StorageHook {
-  const namespace = `${repositoryName}.${pipelineName}`;
+  const {basePath} = React.useContext(AppContext);
+
+  const oldNamespace = `${repoAddress.name}.${pipelineOrJobName}`;
+  const oldData = getStorageDataForNamespace(oldNamespace);
+
+  const namespace = `${basePath}-${repoAddress.location}-${repoAddress.name}-${pipelineOrJobName}`;
   const [version, setVersion] = React.useState<number>(0);
 
   const onSave = (newData: IStorageData) => {
     writeStorageDataForNamespace(namespace, newData);
     setVersion(version + 1); // trigger a React render
   };
 
+  // TODO: Remove this migration logic in a few patches when we know the old namespace is likely no longer being used
+  const oldDataMigrated = React.useRef(false);
+  if (oldData && !oldDataMigrated) {
+    onSave(oldData);
+    window.localStorage.removeItem(oldNamespace);
+  }
```

Comment:
Yeah, I considered that but this still seems like the best overall strategy? If not then we could enumerate over all of the deployments and just copy it over to all of them if they have an entry. Would you prefer that?

Reply:
Yeah, copying to all of them doesn't seem great, and it definitely seems better to remove the old one from `localStorage` once it's been copied over. (Otherwise how would you ever know that it's safe to delete it? At some point you'd have to say it's dead, so even though it's not perfect, I guess it may as well be after that first time it's copied.) @sryza what do you think of that? We copy it over the first time we have a match, and delete it from the old namespace. So it would be effectively blown away for any other `[repo, job]` matches after that first one. Is that better?

Regarding the migration logic, I'm not sure I follow it as written. Nothing sets the ref after localStorage deletion, and it's always going to be truthy since the code doesn't look at `oldDataMigrated.current`.

Reply:
Yeah good catch. I have a commit locally to fix that but was waiting to hear if you wanted to change the logic to enumerate all to include that as well.

Reply:
That sounds reasonable to me